src/peer_manager.rs

```diff
@@ -250,20 +261,34 @@ impl PeerMap {
     }
 }
 
+/// (whether requires resource proof, own group)
+type ExpectJoinResult = Result<(bool, (Prefix<XorName>, Vec<PublicId>)), RoutingTableError>;
```

Comment:
Wouldn't it be better to only name part of that? `(bool, (Prefix<XorName>, Vec<PublicId>))` or `(Prefix<XorName>, Vec<PublicId>)`?